ruoyi-common/src/test/java/com/ruoyi/common/support/ConvertTest.java

```diff
@@ -0,0 +1,202 @@
+package com.ruoyi.common.support;
+
+import java.nio.ByteBuffer;
+
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.charset.Charset;
+import java.util.Set;
+import java.util.TreeSet;
+
+import static com.ruoyi.common.support.Convert.*;
+import static org.junit.Assert.*;
+
+public class ConvertTest {
+
+    @Test
+    public void testToStr() {
+        assertEquals("bar", toStr(null, "bar"));
+        assertEquals("123", toStr(123, "bar"));
+        assertEquals("foo", toStr("foo"));
+    }
+
+    @Test
+    public void testToChar() {
+        assertEquals('b', (char) toChar(null, 'b'));
+        assertEquals('1', (char) toChar(1, 'b'));
+        assertEquals('f', (char) toChar('f'));
+    }
+
+    @Test
+    public void testToByte() {
+        assertEquals((byte) 8, (byte) toByte(null, (byte) 8));
+        assertEquals((byte) 1, (byte) toByte((byte) 1, (byte) 8));
+        assertEquals((byte) 8, (byte) toByte("", (byte) 8));
+        assertEquals((byte) 123, (byte) toByte("123", (byte) 8));
+        assertEquals((byte) 8, (byte) toByte("foo", (byte) 8));
+        assertEquals((byte) 1, (byte) toByte(1));
+    }
+
+    @Test
+    public void testToShort() {
+        assertEquals((short) 8, (short) toShort(null, (short) 8));
+        assertEquals((short) 1, (short) toShort((short) 1, (short) 8));
+        assertEquals((short) 8, (short) toShort("", (short) 8));
+        assertEquals((short) 123, (short) toShort("123", (short) 8));
+        assertEquals((short) 8, (short) toShort("foo", (short) 8));
+        assertEquals((short) 1, (short) toShort(1));
+    }
+
+    @Test
+    public void testToNumber() {
+        assertEquals(8, toNumber(null, 8));
+        assertEquals(8, toNumber("", 8));
+        assertEquals((long) 123, toNumber("123", 8));
+        assertEquals(8, toNumber("foo", 8));
+        assertEquals(1, toNumber(1));
+    }
+
+    @Test
+    public void testToInt() {
+        assertEquals(8, (int) toInt(null, 8));
+        assertEquals(8, (int) toInt("", 8));
+        assertEquals(123, (int) toInt("123", 8));
+        assertEquals(8, (int) toInt("foo", 8));
+        assertEquals(1, (int) toInt(1));
+    }
+
+    @Test
+    public void testToIntArray() {
```

Comment:
I think that instead of testing the length of the arrays crated by the to*Array methods, it would be better to to explicitly create Arrays by hand and assert equality between what you have created, and what the method has returned.

Reply:
ok thanks. Done :)